### ISSUE TITLE
Az.DnsResolver module is not included in the packaged release

### ISSUE THREAD

**Issue by askew** [Azure/azure-powershell#24575]
### Description

Installing `Az` via the downloadable msi installers does n't include the module __DnsResolvers__.

### Issue script & Debug output

```PowerShell
Get-Module -Name Az* -ListAvailable
```

### Environment data

```PowerShell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.4.1
PSEdition                      Core
GitCommitId                    7.4.1
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
Directory: C:\Program Files\WindowsPowerShell\Modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Script     11.5.0                Az                                  Core,Desk
Script     2.17.0                Az.Accounts                         Core,Desk {Disable-AzDataCollection, Disable-AzCo…
Script     2.0.0                 Az.Advisor                          Core,Desk {Disable-AzAdvisorRecommendation, Enabl…
Script     6.0.2                 Az.Aks                              Core,Desk {Disable-AzAksAddOn, Enable-AzAksAddOn,…
Script     1.1.4                 Az.AnalysisServices                 Core,Desk {Resume-AzAnalysisServicesServer, Suspe…
Script     4.0.2                 Az.ApiManagement                    Core,Desk {Add-AzApiManagementApiToGateway, Add-A…
Script     1.0.0                 Az.App                              Core,Desk {Disable-AzContainerAppRevision, Enable…
Script     1.3.0                 Az.AppConfiguration                 Core,Desk {Get-AzAppConfigurationStore, Get-AzApp…
Script     2.2.4                 Az.ApplicationInsights              Core,Desk {Get-AzApplicationInsights, Get-AzAppli…
Script     1.0.0                 Az.ArcResourceBridge                Core,Desk {Get-AzArcResourceBridge, Get-AzArcReso…
Script     2.0.1                 Az.Attestation                      Core,Desk {Add-AzAttestationPolicySigner, Get-AzA…
Script     1.0.0                 Az.Automanage                       Core,Desk {Get-AzAutomanageBestPractice, Get-AzAu…
Script     1.10.0                Az.Automation                       Core,Desk {Export-AzAutomationDscConfiguration, E…
Script     3.5.0                 Az.Batch                            Core,Desk {Remove-AzBatchAccount, Get-AzBatchAcco…
Script     2.0.3                 Az.Billing                          Core,Desk {Get-AzBillingInvoice, Get-AzBillingPer…
Script     3.1.2                 Az.Cdn                              Core,Desk {Clear-AzCdnEndpointContent, Clear-AzFr…
Script     2.0.0                 Az.CloudService                     Core,Desk {Get-AzCloudService, Get-AzCloudService…
Script     1.14.1                Az.CognitiveServices                Core,Desk {Get-AzCognitiveServicesAccount, Get-Az…
Script     7.2.0                 Az.Compute                          Core,Desk {Add-AzImageDataDisk, Add-AzVhd, Add-Az…
Script     1.0.0                 Az.ConfidentialLedger               Core,Desk {Get-AzConfidentialLedger, New-AzConfid…
Script     4.0.0                 Az.ContainerInstance                Core,Desk {Add-AzContainerInstanceOutput, Get-AzC…
Script     4.2.0                 Az.ContainerRegistry                Core,Desk {Connect-AzContainerRegistry, Get-AzCon…
Script     1.14.2                Az.CosmosDB                         Core,Desk {Get-AzCosmosDBAccount, Get-AzCosmosDBA…
Script     1.1.0                 Az.DataBoxEdge                      Core,Desk {Get-AzDataBoxEdgeJob, Get-AzDataBoxEdg…
Script     1.7.1                 Az.Databricks                       Core,Desk {Get-AzDatabricksAccessConnector, Get-A…
Script     1.18.3                Az.DataFactory                      Core,Desk {Add-AzDataFactoryV2DataFlowDebugSessio…
Script     1.0.3                 Az.DataLakeAnalytics                Core,Desk {Get-AzDataLakeAnalyticsDataSource, New…
Script     1.3.0                 Az.DataLakeStore                    Core,Desk {Get-AzDataLakeStoreTrustedIdProvider, …
Script     2.3.0                 Az.DataProtection                   Core,Desk {Backup-AzDataProtectionBackupInstanceA…
Script     1.0.1                 Az.DataShare                        Core,Desk {New-AzDataShareAccount, Get-AzDataShar…
Script     4.3.0                 Az.DesktopVirtualization            Core,Desk {Disconnect-AzWvdUserSession, Expand-Az…
Script     1.1.0                 Az.DevCenter                        Core,Desk {Deploy-AzDevCenterUserEnvironment, Get…
Script     1.0.2                 Az.DevTestLabs                      Core,Desk {Get-AzDtlAllowedVMSizesPolicy, Get-AzD…
Script     1.2.0                 Az.Dns                              Core,Desk {Get-AzDnsRecordSet, New-AzDnsRecordCon…
Script     1.0.1                 Az.ElasticSan                       Core,Desk {Add-AzElasticSanVolumeGroupNetworkRule…
Script     1.6.1                 Az.EventGrid                        Core,Desk {Enable-AzEventGridPartnerTopic, Get-Az…
Script     4.2.1                 Az.EventHub                         Core,Desk {New-AzEventHubAuthorizationRuleSASToke…
Script     1.10.0                Az.FrontDoor                        Core,Desk {New-AzFrontDoor, Get-AzFrontDoor, Set-…
Script     4.0.7                 Az.Functions                        Core,Desk {Get-AzFunctionApp, Get-AzFunctionAppAv…
Script     6.1.0                 Az.HDInsight                        Core,Desk {Add-AzHDInsightClusterIdentity, Add-Az…
Script     2.0.0                 Az.HealthcareApis                   Core,Desk {Get-AzHealthcareApisService, Get-AzHea…
Script     2.7.5                 Az.IotHub                           Core,Desk {Add-AzIotHubKey, Get-AzIotHubEventHubC…
Script     5.2.2                 Az.KeyVault                         Core,Desk {Add-AzKeyVaultCertificate, Add-AzKeyVa…
Script     2.3.0                 Az.Kusto                            Core,Desk {Add-AzKustoClusterLanguageExtension, A…
Script     1.0.0                 Az.LoadTesting                      Core,Desk {Get-AzLoad, New-AzLoad, Remove-AzLoad,…
Script     1.5.1                 Az.LogicApp                         Core,Desk {Get-AzIntegrationAccount, Get-AzIntegr…
Script     1.1.3                 Az.MachineLearning                  Core,Desk {Move-AzMlCommitmentAssociation, Get-Az…
Script     1.0.0                 Az.MachineLearningServices          Core,Desk {Get-AzMLServiceQuota, Get-AzMLServiceU…
Script     1.4.1                 Az.Maintenance                      Core,Desk {Get-AzApplyUpdate, Get-AzConfiguration…
Script     1.2.0                 Az.ManagedServiceIdentity           Core,Desk {Get-AzSystemAssignedIdentity, Get-AzUs…
Script     3.0.0                 Az.ManagedServices                  Core,Desk {Get-AzManagedServicesAssignment, Get-A…
Script     2.0.0                 Az.MarketplaceOrdering              Core,Desk {Get-AzMarketplaceTerms, Invoke-AzMarke…
Script     1.1.2                 Az.Media                            Core,Desk {Sync-AzMediaServiceStorageKey, Set-AzM…
Script     2.3.0                 Az.Migrate                          Core,Desk {Get-AzMigrateDiscoveredServer, Get-AzM…
Script     5.1.1                 Az.Monitor                          Core,Desk {Add-AzLogProfile, Add-AzMetricAlertRul…
Script     1.1.1                 Az.MySql                            Core,Desk {Get-AzMySqlConfiguration, Get-AzMySqlC…
Script     7.4.1                 Az.Network                          Core,Desk {Add-AzApplicationGatewayAuthentication…
Script     1.0.0                 Az.NetworkCloud                     Core,Desk {Deploy-AzNetworkCloudCluster, Disable-…
Script     1.0.0                 Az.Nginx                            Core,Desk {Get-AzNginxCertificate, Get-AzNginxCon…
Script     1.1.2                 Az.NotificationHubs                 Core,Desk {Get-AzNotificationHub, Get-AzNotificat…
Script     3.2.0                 Az.OperationalInsights              Core,Desk {New-AzOperationalInsightsAzureActivity…
Script     1.6.5                 Az.PolicyInsights                   Core,Desk {Get-AzPolicyAttestation, Get-AzPolicyE…
Script     1.1.0                 Az.PostgreSql                       Core,Desk {Get-AzPostgreSqlConfiguration, Get-AzP…
Script     2.0.0                 Az.PowerBIEmbedded                  Core,Desk {Resume-AzPowerBIEmbeddedCapacity, Susp…
Script     1.0.4                 Az.PrivateDns                       Core,Desk {Get-AzPrivateDnsZone, Remove-AzPrivate…
Script     6.8.0                 Az.RecoveryServices                 Core,Desk {Add-AzRecoveryServicesAsrReplicationPr…
Script     1.9.0                 Az.RedisCache                       Core,Desk {Export-AzRedisCache, Get-AzRedisCache,…
Script     1.2.0                 Az.RedisEnterpriseCache             Core,Desk {Export-AzRedisEnterpriseCache, Get-AzR…
Script     2.0.0                 Az.Relay                            Core,Desk {Get-AzRelayAuthorizationRule, Get-AzRe…
Script     1.2.0                 Az.ResourceMover                    Core,Desk {Add-AzResourceMoverMoveResource, Get-A…
Script     6.16.1                Az.Resources                        Core,Desk {Export-AzResourceGroup, Export-AzTempl…
Script     1.6.1                 Az.Security                         Core,Desk {Add-AzSecurityAdaptiveNetworkHardening…
Script     3.1.1                 Az.SecurityInsights                 Core,Desk {Get-AzSentinelAlertRule, Get-AzSentine…
Script     3.1.0                 Az.ServiceBus                       Core,Desk {New-AzServiceBusAuthorizationRuleSASTo…
Script     3.3.2                 Az.ServiceFabric                    Core,Desk {Add-AzServiceFabricClientCertificate, …
Script     2.0.0                 Az.SignalR                          Core,Desk {New-AzSignalR, Get-AzSignalR, Get-AzSi…
Script     4.14.0                Az.Sql                              Core,Desk {Add-AzSqlDatabaseToFailoverGroup, Add-…
Script     2.2.0                 Az.SqlVirtualMachine                Core,Desk {Get-AzAvailabilityGroupListener, Get-A…
Script     2.3.1                 Az.StackHCI                         Core,Desk {Add-AzStackHCIVMAttestation, Disable-A…
Script     1.0.2                 Az.StackHCIVM                       Core,Desk {Add-AzStackHCIVMVirtualMachineDataDisk…
Script     6.1.3                 Az.Storage                          Core,Desk {Add-AzRmStorageContainerLegalHold, Add…
Script     1.3.0                 Az.StorageMover                     Core,Desk {Get-AzStorageMover, Get-AzStorageMover…
Script     2.1.1                 Az.StorageSync                      Core,Desk {Get-AzStorageSyncCloudEndpoint, Get-Az…
Script     2.0.0                 Az.StreamAnalytics                  Core,Desk {Get-AzStreamAnalyticsCluster, Get-AzSt…
Script     1.0.1                 Az.Support                          Core,Desk {Get-AzSupportProblemClassification, Ge…
Script     3.0.6                 Az.Synapse                          Core,Desk {Add-AzSynapseDataFlowDebugSessionPacka…
Script     1.2.2                 Az.TrafficManager                   Core,Desk {Add-AzTrafficManagerCustomHeaderToEndp…
Script     3.2.0                 Az.Websites                         Core,Desk {Add-AzWebAppAccessRestrictionRule, Add…
```

**Comment by Alex-wdy** [comment]
@isra-fel  I added the contact and label.

**Comment by askew** [comment]
I see this is now resolved in release [12.0.0](https://github.com/Azure/azure-powershell/releases/tag/v12.0.0-May2024).
